### PR TITLE
Add grpc query server to hard module

### PR DESCRIPTION
**Pull request by drklee3**
Possible additional / future work in improving the pagination logic, currently pagination requests are only used for calculating page and limit which then is used for the old method of returning a subslice.

### CODE REVIEW

**Review by rhuairahrighairidh — APPROVED** [review]
Approving for testnet.
I wasn't able to find an easy way to disable the public grpc server (without also disabling endpoints needed by ibc relayers). So we should probably revisit grpc generally before mainnet. It would be nice to avoid shipping inconsitent pagination behaviour across modules.

**Review by nddeluca — APPROVED** [review]
lgtm, agree with @rhuairahrighairidh on the pagination too.  I also believe we can move the grpc_query.go to x/hard/qrpc_query.go -- there shouldn't be a need to access private keeper functionality.  The keeper is a lower level dependency.

File: x/hard/keeper/grpc_query.go [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+
+	// Get params
+	params := qs.keeper.GetParams(sdkCtx)
+
+	return &types.QueryParamsResponse{
+		Params: params,
+	}, nil
+}
+
+func (qs QueryServer) Accounts(ctx context.Context, req *types.QueryAccountsRequest) (*types.QueryAccountsResponse, error) {
```
[the remark this answers is not in the document]
Comment (drklee3) [reply x/hard/keeper/grpc_query.go:44]: Thoughts on also removing the name parameter in the `QueryAccountsRequest`? It doesn't seem really necessary if there's only one account, and it should make it more consistent to the cdp accounts query where it also doesn't have any paramters.

**Comment by drklee3** [comment]
> lgtm, agree with @rhuairahrighairidh on the pagination too. I also believe we can move the grpc_query.go to x/hard/qrpc_query.go -- there shouldn't be a need to access private keeper functionality. The keeper is a lower level dependency.

Yeah main reason why I kept it in keeper was for the use of `accountKeeper` and `bankKeeper` fields but those can be made public for access from outside of the keeper if preferred?

I'm also trying to integrate Cosmos's `FilteredPaginate` so we can properly use the `PageRequest` and respond with a valid `PageResponse` but it's kind of tricky for some queries that use different keys since the `FilteredPaginate` method takes a store as an argument -- I'm assuming we don't want to iterate over entire stores instead of using specific stores we have that are more optimized.

Will do in a separate pr if it turns out okay since it would need a decent amount more refactoring.  There was some things like filtering CDPs that are a lot more complicated to do with FilteredPaginate that probably need more significant work and testing to use different stores.

**Comment by nddeluca** [comment]
For the keepers, it'd be better to pass them from the AppModule and directly instantiate the querier with those dependencies -- ex something like ` keeper.NewQueryServerImpl(am.keeper, am.accountKeeper, am.bankKeeper)`.

And good point on the FilteredPaginate, it'd be good to find a consistent way to do this across the modules -- it makes sense to spike that in a different changeset

**Comment by drklee3** [comment]
Oh yeah that's a lot cleaner, thanks!

**Pull request merged**